### PR TITLE
[AMORO-2341] Unify the naming format for threads

### LINKED ISSUE #2341: [Improvement]: Unify the naming format for Threads
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

We are preparing to unify the thread naming in Amoro according to the following specifications:
 - in lowercase and use hyphens(-) to separate different words.

For example, `renew-blocker-thread-1` is a valid thread name, instead of 
```java
ThreadFactory threadFactory =
              new ThreadFactoryBuilder()
                  .setDaemon(true)
                  .setNameFormat("Renew Blocker Thread %d")
                  .build();
```

### How should we improve?

Find and modify all non-standard thread names in Amoro.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

Close #2341.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java [hunk 1]
```diff
@@ -357,7 +357,7 @@ private TServer createThriftServer(
   private ThreadFactory getThriftThreadFactory(String processorName) {
     return new ThreadFactoryBuilder()
         .setDaemon(false)
-        .setNameFormat("thrift-server-" + processorName + "-%d")
+        .setNameFormat("thrift-server-" + StringUtils.lowerCase(processorName) + "-%d")
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java:360]: We can translate the camel-style process name to lowercase one by:
```
String.join("-", StringUtils.splitByCharacterTypeCamelCase(processorName)).toLowerCase(Locale.ROOT)
```

File: ams/server/src/main/java/com/netease/arctic/server/table/executor/BaseTableExecutor.java [hunk 2]
```diff
@@ -50,7 +51,7 @@ protected BaseTableExecutor(TableManager tableManager, int poolSize) {
             poolSize,
             new ThreadFactoryBuilder()
                 .setDaemon(false)
-                .setNameFormat("ASYNC-" + getThreadName() + "-%d")
+                .setNameFormat("async-" + StringUtils.lowerCase(getThreadName()) + "-%d")
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/table/executor/BaseTableExecutor.java:54]: Same here.

**Review by zhoujinsong — COMMENTED** [review]
@tcodehuber Thanks a lot for your wrok!
I left some small suggestion.

File: ams/server/src/main/java/com/netease/arctic/server/table/executor/BaseTableExecutor.java (hunk 2 above)
Comment (tcodehuber) [reply ams/server/src/main/java/com/netease/arctic/server/table/executor/BaseTableExecutor.java:54]: Done, thanks for your useful suggestion.

File: ams/server/src/main/java/com/netease/arctic/server/table/executor/BaseTableExecutor.java [hunk 3]
```diff
@@ -50,7 +52,12 @@ protected BaseTableExecutor(TableManager tableManager, int poolSize) {
             poolSize,
             new ThreadFactoryBuilder()
                 .setDaemon(false)
-                .setNameFormat("ASYNC-" + getThreadName() + "-%d")
+                .setNameFormat(
+                    "async-"
+                        + String.join(
+                                "-", StringUtils.splitByCharacterTypeCamelCase(getThreadName()))
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/table/executor/BaseTableExecutor.java:58]: How about moving the translate codes into the `getThreadName` method, as we may want the return value have meet the thread name requirement?

Comment (tcodehuber) [reply ams/server/src/main/java/com/netease/arctic/server/table/executor/BaseTableExecutor.java]: Done.

File: ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java [hunk 4]
```diff
@@ -69,11 +69,7 @@
 import java.net.InetSocketAddress;
 import java.nio.file.Files;
 import java.nio.file.Paths;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (tcodehuber) [reply ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java]: OK
